seed/energy/meter_data_processor/monthly_data_aggregator.py

```diff
@@ -0,0 +1,106 @@
+import datetime
+import json
+import logging
+
+import requests
+from dateutil import tz
+from django.conf import settings
+
+from seed.models import (
+    Meter,
+    TimeSeries,
+)
+
+
+_log = logging.getLogger(__name__)
+
+
+def aggr_sum_metric(data, localtzone):
+    '''
+    aggregate monthly data kairos and push it to postgres. data represents the aggregate query
+    '''
+
+    headers = {'content-type': 'application/json'}
+
+    url = settings.TSDB['query_url']
+
+    data = json.dumps(data)
+
+    r = requests.post(url, data=data, headers=headers)
+    # length of output array. Should be 1 per group since it's monthly aggregation and we are querying only for one month
+
+    if 'queries' not in r.json():
+        return
+
+    res = r.json()['queries'][0]['results']
+
+    length = len(res)
+
+    # Retrieve required values from array and call posgres insert function
+    for num in range(0, length):
+        if len(res[num]['tags']) > 0:
+            res_tags = res[num]['tags']
+
+            gb_bldg_canonical_id = res_tags['canonical_id'][0]
+            gb_mtr_id = res_tags['custom_meter_id'][0]
+            gb_energy_type_id = res_tags['energy_type_int'][0]
+
+            resultarrlen = len(res[num]['values'])
+            res_values = res[num]['values']
+
+            for numv in range(0, resultarrlen):
+                gb_timestamp = res_values[numv][0]
+                gb_agg_reading = res_values[numv][1]
+
+                timestamp = datetime.datetime.fromtimestamp(gb_timestamp / 1000.0, tz.gettz(localtzone))
+                tsMonthStart = timestamp.replace(day=1).replace(hour=0).replace(minute=0).replace(second=0)
+                tsMonthEnd = timestamp.replace(hour=23).replace(minute=59).replace(second=59)
+
+                # NL: I hard coded this here because year is not defined. I do not know from where it should be coming
+                year = 2010
```

Comment:
The year is tsMonthEnd.year, I will use calendar.isleap to check the leap
year.

On Thu, Feb 11, 2016 at 4:54 PM, Nicholas Long <notifications@github.com>
wrote:

> In seed/energy/meter_data_processor/monthly_data_aggregator.py
> <https://github.com/SEED-platform/seed/pull/714#discussion_r52673373>:
>
> > +            gb_mtr_id = res_tags['custom_meter_id'][0]
> > +            gb_energy_type_id = res_tags['energy_type_int'][0]
> > +
> > +            resultarrlen = len(res[num]['values'])
> > +            res_values = res[num]['values']
> > +
> > +            for numv in range(0, resultarrlen):
> > +                gb_timestamp = res_values[numv][0]
> > +                gb_agg_reading = res_values[numv][1]
> > +
> > +                timestamp = datetime.datetime.fromtimestamp(gb_timestamp / 1000.0, tz.gettz(localtzone))
> > +                tsMonthStart = timestamp.replace(day=1).replace(hour=0).replace(minute=0).replace(second=0)
> > +                tsMonthEnd = timestamp.replace(hour=23).replace(minute=59).replace(second=59)
> > +
> > +                # NL: I hard coded this here because year is not defined. I do not know from where it should be coming
> > +                year = 2010
>
> year was not defined in this method. I hard coded it to 2010... where
> should this value come from?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SEED-platform/seed/pull/714/files#r52673373>.
>